tests/beacon-test.rs

```diff
@@ -0,0 +1,391 @@
+    }
+
+    for i in 0..beacons.len() {
+        assert_eq!(beacons[i], deserialized[i]);
+    }
+}
+
+/// Computes the amount of time needed to randomly generate one beacon
+#[test]
+pub fn check_generation_time() {
+    let now = SystemTime::now();
+    println!(
+        "Starting beacon creation at {}ns",
+        now.elapsed().unwrap().as_nanos()
+    );
+    let _beacon = rnd_beacon();
+    let elapsed = now.elapsed().unwrap();
+    println!(
+        "Beacon creation finished after {}ns, or {}s",
+        elapsed.as_nanos(),
+        elapsed.as_secs()
+    );
+}
+
+/// Computes the amount of time needed to generate 5000 random beacons
+#[test]
+pub fn check_time_for_generation_of_5000_beacons() {
+    let now = SystemTime::now();
+    let first = now.elapsed().unwrap().as_nanos();
+    println!("Starting beacon creation at {}ns", first);
+    for _i in 0..5001 {
+        rnd_beacon();
+    }
+    let elapsed = now.elapsed().unwrap();
+    let second = elapsed.as_nanos();
+    let secs = elapsed.as_secs();
+    println!(
+        "Beacon creation finished at {}ns, or {}s.\nIt took {}ns to create 5000 beacons",
+        second,
+        secs,
+        second - first
+    );
+}
+
+/// Computes the amount of time needed to serialize 5000 beacons
+#[test]
+pub fn check_time_for_serialization_of_5000_beacons() {
+    let now = SystemTime::now();
+    let mut beacons: Vec<Beacon> = Vec::new();
+    let startc = now.elapsed().unwrap().as_nanos();
+    println!("Beginning creation of beacons after {}ns", startc);
+    for _i in 0..5001 {
+        beacons.push(rnd_beacon());
+    }
+    let afterc = now.elapsed().unwrap().as_nanos();
+    println!("Beacon creation finished after {}ns", afterc - startc);
+
+    let starts = now.elapsed().unwrap().as_nanos();
+    println!("Starting serialization after {}ns", starts);
+    for x in beacons {
+        serde_cbor::to_vec(&x).expect("A problem occured while serializing");
+    }
+    let afters = now.elapsed().unwrap().as_nanos();
+
+    println!("Serializing 5000 Beacons took {}ns", afters - starts);
+}
+
+/// Computes the amount of time needed to deserialize 5000 beacons
+#[test]
+pub fn check_time_for_deserialization_of_5000_beacons() {
+    let mut beacons: Vec<Beacon> = Vec::new();
+    let mut serialized: Vec<Vec<u8>> = Vec::new();
+
+    for _i in 0..5001 {
+        beacons.push(rnd_beacon());
+    }
+
+    for x in beacons {
+        serialized.push(serde_cbor::to_vec(&x).expect("A problem occured while serializing"));
+    }
+    let now = SystemTime::now();
+    let startd = now.elapsed().unwrap().as_nanos();
+    println!("Starting deserialization after: {}ns", startd);
+    for x in serialized {
+        serde_cbor::from_slice::<Beacon>(&x).expect("Something bad happened");
+    }
+    let afterd = now.elapsed().unwrap().as_nanos();
+    println!("Deserialization finished after: {}ns", afterd - startd);
+}
+
+#[test]
+#[ignore]
```

Comment:
why is this test and the following ignored? please add a comment

Reply:
Added comments for why those tests are ignored